### PR TITLE
Water-3070 Charge Version workflow tweaks

### PR DESCRIPTION
**Pull request by Stephan-Hough**
**Feat/water 3070 cv wf tweaks**

- Charge version workflow ID added to the session data key
- Charge version workflow ID has been added as optional query param to all the charge info and charge workflow routes. 
- User can now create as many charge version workflows from the licences page as the licence version is set to null in the charge version workflow
- The charge version work flow "To set up” tab now includes the licence version start date
- The charge version work flow “Review” tab now includes the charge version start date
- Users can now create charge versions for expired/future dated licences

### CODE REVIEW

**Review by DGajewska — CHANGES_REQUESTED** [review]

File: src/internal/views/nunjucks/charge-information/workflow-tabs/to-set-up.njk [hunk 1]
```diff
@@ -15,18 +15,18 @@
                 <tr class="govuk-table__row">
                     <th class="govuk-table__header" scope="col">Licence</th>
                     <th class="govuk-table__header" scope="col">Licence holder</th>
-                    <th class="govuk-table__header" scope="col">Licence start date</th>
-                    <th class="govuk-table__header" scope="col">Action</th>
+                    <th class="govuk-table__header table__header--width-20" scope="col">Licence version <br> start date</th>
+                    <th class="govuk-table__header table__header--width-20" scope="col">Action</th>
```
[the remark this answers is not in the document]
Comment (Stephan-Hough) [reply src/internal/views/nunjucks/charge-information/workflow-tabs/to-set-up.njk:19]: Yes - I had to add this here to make sure the table looked more readable. This is something that was requested by UCD. The other tabs was signed off as it by UCD.

File: src/internal/views/nunjucks/charge-information/workflow-tabs/review.njk [hunk 2]
```diff
@@ -17,7 +17,7 @@
                     <th class="govuk-table__header" scope="col">Licence</th>
                     <th class="govuk-table__header" scope="col">Licence holder</th>
                     <th class="govuk-table__header" scope="col">Set up by</th>
-                    <th class="govuk-table__header" scope="col">Charge information start date</th>
+                    <th class="govuk-table__header" scope="col">Charge <br> information <br> start date</th>
```
[the remark this answers is not in the document]
Comment (Stephan-Hough) [reply src/internal/views/nunjucks/charge-information/workflow-tabs/review.njk:20]: This is something that was requested by UCD.

File: src/internal/modules/charge-information/lib/routing.js [hunk 3]
```diff
@@ -1,27 +1,29 @@
 'use strict';
 
 const queryString = require('querystring');
+const { isEmpty } = require('lodash');
 
-const createUrl = urlTail => licenceId => {
-  return `/licences/${licenceId}/charge-information/${urlTail}`;
+const cleanObject = obj => {
+  for (const key in obj) {
+    if (obj[key] === undefined || ((obj[key]).length) < 1) {
```
[the remark this answers is not in the document]
Comment (Stephan-Hough) [reply src/internal/modules/charge-information/lib/routing.js]: we discussed and came up with an agreed solution. Thanks

File: src/internal/modules/charge-information/lib/routing.js [hunk 4]
```diff
@@ -1,27 +1,29 @@
...
+  }
+  return isEmpty(obj) ? null : obj;
 };
 
-exports.getChargeElementStep = (licenceId, elementId, step) => {
-  return createUrl(`charge-element/${elementId}/${step}`)(licenceId);
+const createUrl = urlTail => (licenceId, queryParams = null) => {
+  const url = `/licences/${licenceId}/charge-information/${urlTail}`;
+  const qp = cleanObject(queryParams);
+  return isEmpty(qp) ? url : `${url}?${queryString.stringify(qp)}`;
```
[the remark this answers is not in the document]
Comment (Stephan-Hough) [reply src/internal/modules/charge-information/lib/routing.js:18]: removed in the cleanObject method

**Review by DGajewska — APPROVED** [review]

**Pull request merged**